drivers/hdlc_rcp_if/hdlc_rcp_if_uart.c

```diff
@@ -0,0 +1,216 @@
+ */
+
+/**
+ * @file
+ * IEEE 802.15.4 HDLC RCP interface - serial communication interface (UART)
+ */
+
+/* -------------------------------------------------------------------------- */
+/*                                  Includes                                  */
+/* -------------------------------------------------------------------------- */
+#include <zephyr/drivers/uart.h>
+#include <openthread/platform/radio.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <zephyr/init.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/net/hdlc_rcp_if/hdlc_rcp_if.h>
+#include <zephyr/net/ieee802154_radio.h>
+#include <zephyr/net/openthread.h>
+#include <zephyr/sys/ring_buffer.h>
+
+/* -------------------------------------------------------------------------- */
+/*                                  Definitions                               */
+/* -------------------------------------------------------------------------- */
+
+#define DT_DRV_COMPAT uart_hdlc_rcp_if
+
+#define LOG_MODULE_NAME hdlc_rcp_if_uart
+#define LOG_LEVEL       CONFIG_HDLC_RCP_IF_DRIVER_LOG_LEVEL
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+struct openthread_uart {
+	struct k_work work;
+	struct ring_buf *rx_ringbuf;
+	const struct device *dev;
+	atomic_t tx_busy;
+
+	hdlc_rx_callback_t cb;
+	void *param;
+};
+
+#define OT_UART_DEFINE(_name, _ringbuf_size) \
+	RING_BUF_DECLARE(_name##_rx_ringbuf, _ringbuf_size); \
+	static struct openthread_uart _name = { \
+		.rx_ringbuf = &_name##_rx_ringbuf, \
+	}
+OT_UART_DEFINE(ot_uart, CONFIG_OPENTHREAD_HDLC_RCP_IF_UART_RING_BUFFER_SIZE);
+
+struct ot_hdlc_rcp_context {
+	struct net_if *iface;
+	struct openthread_context *ot_context;
+};
+
+static const uint8_t *write_buffer;
+static uint16_t write_length;
+
+/* -------------------------------------------------------------------------- */
+/*                             Private functions                              */
+/* -------------------------------------------------------------------------- */
+
+static void ot_uart_rx_cb(struct k_work *item)
+{
+	struct openthread_uart *otuart =
+		CONTAINER_OF(item, struct openthread_uart, work);
+	static uint8_t *data;
+	uint32_t len;
+
+	len = ring_buf_get_claim(otuart->rx_ringbuf, &data,
+				 otuart->rx_ringbuf->size);
+	if (len > 0) {
+		otuart->cb(data, len, otuart->param);
+		ring_buf_get_finish(otuart->rx_ringbuf, len);
+	}
+}
+
+static void uart_tx_handle(const struct device *dev)
+{
+	uint32_t len;
+
+	if (write_length) {
+		len = uart_fifo_fill(dev, write_buffer, write_length);
+		write_buffer += len;
+		write_length -= len;
+	} else {
+		uart_irq_tx_disable(dev);
+		atomic_clear(&ot_uart.tx_busy);
+	}
+}
+
+static void uart_rx_handle(const struct device *dev)
+{
+	uint32_t rd_len = 0, len;
+	uint8_t *data;
+
+	len = ring_buf_put_claim(
+		ot_uart.rx_ringbuf, &data,
+		ot_uart.rx_ringbuf->size);
+	if (len > 0) {
+		rd_len = uart_fifo_read(dev, data, len);
+
+		int err = ring_buf_put_finish(
+			ot_uart.rx_ringbuf, rd_len);
```

Comment:
This should fit into 80 chars/line limit, I guess no need to break the line.
I'm also a bit puzzled on why compliance doesn't "complain" it typically requires an empty line after variable definition.